### ISSUE TITLE
An "unused variables `p`" problem always exists

### ISSUE THREAD

**Issue by feynhang** [rust-lang/rust-analyzer#17503]
The previous code:
```rust
fn start() -> anyhow::Result<()> {
}
```

**Comment by lnicola** [comment]
I don't understand the issue, but that looks like a `rustc` warning, so it's probably correct. I cloned https://github.com/feynhang/tinyfileshare/, but the code looks different.

**Comment by feynhang** [comment]
Sorry，the code has not been pushed to github yet.  
Problem is just like the picture shown:
![2024-06-27_224556](https://github.com/rust-lang/rust-analyzer/assets/44837656/2c47f942-a5fa-450c-b8c3-76fa463eb020)
![2024-06-27_224611](https://github.com/rust-lang/rust-analyzer/assets/44837656/0366669d-7b5f-4320-af5c-b0260b853db8)

**Comment by lnicola** [comment]
Line 184 seems to be `let remote_listener: TcpListener`? Did you save the file, and do you have any errors under Output / Rust Analyzer Language Server"?

**Comment by feynhang** [comment]
Yes, I saved. And nothing under Output/Rust Analyzer Language Server.
This is indeed a warning from rustc, but there is no problem while `cargo check/clippy` in termianl.

**Comment by lnicola** [comment]
It's hard to say, then. I'll take another look once you push the changes.

**Comment by feynhang** [comment]
I have pushed the code to github, and I cloned the code on another laptop, and then opened the project with vscode, everything is normal, so it seems to be a cache problem, but I have cleaned up the workspace storage, I don't know which caches need to be cleaned up, does rust-analyzer have its own cache directory? Can it be cleaned up?

**Comment by feynhang** [comment]
Indeed, as you said, after cargo clean and restarting vscode, the warning disappeared.
I didn't think that the cache of rust-analyzer can also be cleared by cargo clean.
However, thank you very much.

**Comment by lnicola** [comment]
rust-analyzer has no cache, restarting will reset its state. VS Code has does some caching, which causes problems when you rename a file and only change the case (`Foo.rs` to `foo.rs`). That one you have to clear manually.

**Comment by feynhang** [comment]
In general, I understand`cargo clean`as the cache used when`cargo build` is executed, thank you for letting me understand that it is related to resetting the rust-analyzer state. 
In addition, I still don't know what caused the rust-analyzer to get stuck, but it would be helpful to have a simple and effective way to solve this problem.

**Comment by lnicola** [comment]
> thank you for letting me understand that it is related to resetting the rust-analyzer state

It's not, except some proc macro stuff. But that was a `rustc` error, and `cargo` sometimes gets confused, so `cargo clean` can fix that. I still suspect RA in your case, because of this:

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/1f212ee5-f447-4843-a50d-09f25d3031dd)

But there's no way to figure out what happened.
